openssl-sys/src/openssl_shim.c

```diff
@@ -3,6 +3,33 @@
 #include <openssl/dh.h>
 #include <openssl/bn.h>
 
+#if defined(__APPLE__) || defined(__linux)
+
+#include<pthread.h>
+#include<openssl/crypto.h>
+
+unsigned long thread_id()
+{
+    unsigned long ret = (unsigned long)pthread_self();
+    return ret;
+}
+
+void rust_openssl_set_id_callback() {
+  CRYPTO_set_id_callback((unsigned long (*)())thread_id);
```

Comment:
This is what the openssl testing code did, and I'm scared of function pointers, so I did it too :smile: